addon/-private/system/model/internal-model.js

```diff
@@ -583,10 +583,19 @@ export default class InternalModel {
     let changedKeys;
 
     if (this.hasRecord) {
-      changedKeys = this._changedKeys(data.attributes);
+      if (typeof this._record._changedKeys === 'function') {
+        changedKeys = this._record._changedKeys(data.attributes);
+      } else {
+        changedKeys = this._changedKeys(data.attributes);
+      }
+    }
+
+    if (this.hasRecord && this._record._assignAttributes) {
```

Comment:
I'm checking to see if this is actually necessary; might drop it if not otherwise i agree i'll switch to the other pattern